src/SI_Toolkit/Predictors/predictor_wrapper.py

```diff
@@ -0,0 +1,136 @@
+import os
+import yaml
+from copy import deepcopy as dcp
+
+# predictors config
+predictors_config = yaml.load(open(os.path.join('SI_Toolkit_ASF', 'config_predictors.yml'), 'r'), Loader=yaml.FullLoader)
+
+
+class PredictorWrapper:
+    def __init__(self):
+
+        self.horizon = None
+        self.batch_size = None
+
+        self.predictor = None
+
+        self.predictors_config = dcp(predictors_config['predictors'])  # TODO: This must be read only!
+        self.predictor_name_main: str = predictors_config['predictor_name_main']
+
+        self.predictor_name: str = self.predictor_name_main
+        self.predictor_config = dcp(self.predictors_config[self.predictor_name])
+        self.predictor_type: str = self.predictor_config['predictor_type']
+        self.model_name: str = self.predictor_config['model_name']
+
+
+    def configure(self, batch_size, horizon, predictor_specification=None, compile_standalone=False):
+
+        self.update_predictor_config_from_specification(predictor_specification)
+
+        compile_standalone = {'disable_individual_compilation': not compile_standalone}
+
+        self.batch_size = batch_size
+        self.horizon = horizon
+
+        if self.predictor_type == 'neural':
+            from SI_Toolkit.Predictors.predictor_autoregressive_tf import predictor_autoregressive_tf
+            self.predictor = predictor_autoregressive_tf(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config, **compile_standalone)
+
+        elif self.predictor_type == 'GP':
+            from SI_Toolkit.Predictors.predictor_autoregressive_GP import predictor_autoregressive_GP
+            self.predictor = predictor_autoregressive_GP(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config, **compile_standalone)
+
+        elif self.predictor_type == 'ODE':
+            from SI_Toolkit.Predictors.predictor_ODE import predictor_ODE
+            self.predictor = predictor_ODE(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config)
+
+        elif self.predictor_type == 'ODE_TF':
+            from SI_Toolkit.Predictors.predictor_ODE_tf import predictor_ODE_tf
+            self.predictor = predictor_ODE_tf(horizon=self.horizon, batch_size=self.batch_size, **self.predictor_config, **compile_standalone)
+
+        else:
+            raise NotImplementedError('Type of the predictor not recognised.')
+
+    def configure_with_compilation(self, batch_size, horizon, predictor_specification=None):
+        """
+        To get max performance
+        use this for standalone predictors (like in Brunton test)
+        but not predictors within controllers
+        """
+        self.configure(batch_size, horizon, compile_standalone=True)
```

Comment:
Conceptually I'm unsure but better to ask: `predictor_specification` is missing from the arguments here. Is that intended?

Reply:
Yes, you are right. Corrected.